src/error.rs

```diff
@@ -0,0 +1,125 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::fmt;
+use std::result;
+
+use failure::{Backtrace, Context, Fail};
+use rocket::http::{Status};
+use rocket::response::{Responder, Response};
+use rocket::{self, response, Request};
+use rocket_contrib::Json;
+
+use logging::MozlogLogger;
+
+pub type HandlerResult<T> = result::Result<T, HandlerError>;
+
+#[derive(Debug)]
+pub struct HandlerError {
+    inner: Context<HandlerErrorKind>,
+}
+
+#[derive(Clone, Eq, PartialEq, Debug, Fail)]
+pub enum HandlerErrorKind {
+    /// 400 Bad Request
```

Comment:
do we need to handle `401`s ? (`401 unauthorized`)

Reply:
Not yet, there's no authentication in the first cut of this service, so we never return a 401.